### PR TITLE
Remove deprecated Orbit.from_body_ephem and its tests

### LINKED ISSUE #1095: Remove internal uses of deprecated Orbit.from_body_ephem
Doing a quick search, the only place where we are using `Orbit.from_body_ephem` is this one:

https://github.com/poliastro/poliastro/blob/6ee8e9386a27d4d241cd0e9cb22f6fb6fbf2cf45/src/poliastro/plotting/_base.py#L372-L374

Instead, we should instead be doing this:

```python
    ephem = Ephem.from_body(body, J2000_TDB)
    orbit = Orbit.from_ephem(Sun, ephem, J2000_TDB)
```

And, after that, tests covering `Orbit.from_body_ephem` can be removed.

### PR DESCRIPTION
**Pull request by ishanSrt**
Closes #1095

### CODE REVIEW

**Comment by ishanSrt** [comment]
should I add a decorator to the codecov `exclude_lines` and add it to the deprecated method? @astrojuanlu

**Comment by ishanSrt** [comment]
Should I add a decorator to the deprecated function and add it to the `exclude_lines` for `coveragerc` @astrojuanlu

**Comment by astrojuanlu** [comment]
Hello @ishanSrt , sorry for the delay! Are you able to see the Quality failures? In the logs, I see this:

```
2021-02-14T17:19:12.4724784Z check run-test: commands[0] | flake8 src tests
2021-02-14T17:19:16.2979791Z tests/tests_twobody/test_orbit.py:21:1: F401 'poliastro.bodies.Pluto' imported but unused
2021-02-14T17:19:16.2980742Z tests/tests_twobody/test_orbit.py:40:1: F401 'poliastro.frames.equatorial.ICRS' imported but unused
2021-02-14T17:19:16.2981616Z tests/tests_twobody/test_orbit.py:55:1: F401 'poliastro.warnings.TimeScaleWarning' imported but unused
2021-02-14T17:19:16.3174916Z ERROR: InvocationError for command /home/vsts/work/1/s/.tox/check/bin/flake8 src tests (exited with code 1)
```

please address this proactively so we can focus on the interesting parts of the review :)

**Comment by astrojuanlu** [comment]
Interestingly, the image tests failed, which suggests that there are some pixel differences. @ishanSrt do you know how to investigate those?

**Comment by ishanSrt** [comment]
I don't @astrojuanlu but can you point me to some documentation? and I'll look

File: tests/tests_twobody/test_orbit.py [hunk 1]
```diff
@@ -52,11 +50,7 @@
 from poliastro.frames.util import get_frame
 from poliastro.twobody.angles import E_to_M, nu_to_E
 from poliastro.twobody.orbit import Orbit
-from poliastro.warnings import (
-    OrbitSamplingWarning,
-    PatchedConicsWarning,
-    TimeScaleWarning,
-)
+from poliastro.warnings import OrbitSamplingWarning, PatchedConicsWarning
```
Comment (ishanSrt) [inline tests/tests_twobody/test_orbit.py:53]: also not a fan of these kinds of refactoring changes as they add problems later to the git blame 😅, but the quality checks won't allow it for now.

**Comment by ishanSrt** [comment]
> Interestingly, the image tests failed, which suggests that there are some pixel differences. @ishanSrt do you know how to investigate those?

also they aren't faling locally on my machine, I'll have to add the CI to my repo and spit the output out somewhere, for debugging

File: tests/tests_twobody/test_orbit.py (hunk 1 above)
Comment (astrojuanlu) [reply tests/tests_twobody/test_orbit.py:53]: the PEP8 / Black deity rules 😈

**Comment by astrojuanlu** [comment]
@ishanSrt If they're failing on CI but not locally, perhaps you have to rebuild your tox environment. Something like:

```
$ tox -e -r py38-images
```

(only do `-r` once to save time)

Can you confirm whether this passes or breaks?

**Comment by astrojuanlu** [comment]
Also, this will need a rebase

**Comment by ishanSrt** [comment]
@astrojuanlu can't make any of the three tests fail locally. tried with different python versions and rebuilding the tox environment.

**Comment by ishanSrt** [comment]
> Should I add a decorator to the deprecated function and add it to the `exclude_lines` for `coveragerc` @astrojuanlu ?

@astrojuanlu please review

**Review by astrojuanlu — APPROVED** [review]
Thanks for this @ishanSrt and sorry for the long delay 🙏🏽 This can be merged already.

**Pull request merged**